File: problem012/answer012.py
#!/usr/bin/python
"""chicks' answer to Euler Project problem #12"""

def factor(n):
	factors = [1]
	x = 2
	while x <= n:
		if n % x == 0:
			factors.append(x)
		if (x % 10000000) == 0:
			print "\tchecking " + str(x) + " for " + str(n) + " (" + str(float(x)/n) + "%)"
		x = x + 1

#	print str(n) + " has " + str(len(factors)) + " factors"

	return len(factors)

max_divisors = 0
n = 0
triangle = 0

while max_divisors < 502:
	n += 1
	loop_n = n
	triangle = 0
	while loop_n:
		triangle += loop_n
		loop_n -= 1
	divisors = factor(triangle)

	if divisors > max_divisors:
		max_divisors = divisors
		print "n=" + str(n) + " triangle " + str(triangle) + " has " + str(max_divisors) + " divisors"


print "ANSWER:" + str(triangle)

File: problem012/answer012.pl
#!/usr/bin/perl

use strict;
use warnings;
use Math::Prime::Util qw(divisors);
use Data::Dumper;


sub count_factors {
	my ($n) = @_;

	my @factors = divisors($n);
#	print "$n: " . join(',', @factors) . "\n";
	my $count = scalar @factors;

	return $count;
}

my $max_divisors = 0;
my $n = 0;
my $triangle = 0;

while ($max_divisors < 502) {
	$n++;
	my $loop_n = $n;
	$triangle = 0;

	while($loop_n) {
		$triangle += $loop_n;
		$loop_n --;
	}

	my $divisors = count_factors($triangle);
	if ($divisors > $max_divisors) {
		$max_divisors = $divisors;
		print "n=$n triangle=$triangle has $max_divisors divisors\n";
	}
}

print "ANSWER:$triangle\n";

__END__
		print "n=" + str(n) + " triangle " + str(triangle) + " has " + str(max_divisors) + " divisors"

